dashboard/app/controllers/home_controller.rb

```diff
@@ -35,12 +35,14 @@ def health_check
 
   GALLERY_PER_PAGE = 5
 
-  # Signed in student with course progress: redirect course overview page
-  # Signed in, not student with assigned course: redirect to /home
+  # Signed in student, with an assigned course/script: redirect to course overview page
+  # Note: the student will be redirected to the course or script in which they
+  # most recently made progress, which may not be an assigned course or script.
+  # Signed in student or teacher, without an assigned course/script: redirect to /home
   # Signed out: redirect to /courses
   def index
     if current_user
-      if current_user.student? && current_user.primary_script && !session['clever_link_flag']
+      if current_user.student? && current_user.assigned_course_or_script? && !session['clever_link_flag']
```

Comment:
It was redundant to ask check here for primary_script because it the user has an assigned_course_or_script, they will have a primary_script.

Reply:
Ugh, actually maybe not. Fixed in #18572 18572